### PR TITLE
[expo-notifications] Not starting an extra thread in the background service

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Currently in `ExpoFcmMessagingService` was started a new thread(actually it is the main thread 🤷🏼‍♂️) to do the async job. It's not needed and may lead to bigger problems in the future. I think we should make it sync. 

See also https://stackoverflow.com/questions/41067081/does-firebasemessagingservice-run-in-the-background-by-default/41067993#41067993

# How

Make it synchronous ;)
Before:
```
host.exp.exponent E/host.exp.exponent: onMessageReceived
host.exp.exponent E/host.exp.exponent: Thread[Firebase-ExpoFcmMessagingService,5,main]
host.exp.exponent E/host.exp.exponent: onMessageReceived in API
host.exp.exponent E/host.exp.exponent: Thread[main,5,main]
host.exp.exponent E/host.exp.exponent: loading icon
host.exp.exponent E/host.exp.exponent: Thread[main,5,main]
```
Now:
```
host.exp.exponent E/host.exp.exponent: onMessageReceived
host.exp.exponent E/host.exp.exponent: Thread[Firebase-ExpoFcmMessagingService,5,main]
host.exp.exponent E/host.exp.exponent: onMessageReceived in API
host.exp.exponent E/host.exp.exponent: Thread[Firebase-ExpoFcmMessagingService,5,main]
host.exp.exponent E/host.exp.exponent: loading icon
host.exp.exponent E/host.exp.exponent: Thread[Firebase-ExpoFcmMessagingService,5,main]
```

I don't know if we should apply it to the legacy notification too.

# Test Plan

- sending notification via https://expo.io/notifications when the app is backgrounded/kill.

### CODE REVIEW

File: android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java [hunk 1]
```diff
@@ -40,44 +39,36 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
...
-      }
-
-      @Override
-      public void onFailure() {
-        Log.w("expo-notifications", "Couldn't get experience from remote message.", null);
+        // If it's an older experience or useNextNotificationsApi is set to false, let's use the legacy notifications API
         dispatchToLegacyNotificationModule(remoteMessage);
+      } catch (JSONException e) {
+        e.printStackTrace();
+        EXL.e("expo-notifications", "No experience found for id " + remoteMessage.getData().get("experienceId"));
```
Comment (sjchmiela) [inline android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java:69]: Actually, don't we guard against `No experience found` situation with `if (experienceDBObject != null)`? I mean… if we are inside the `if` doesn't it mean an experience has been found?

File: android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java [hunk 2]
```diff
@@ -40,44 +39,36 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
...
       }
-    });
+    }
```
Comment (sjchmiela) [inline android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java:71]: In the old code when no experience was being found we still dispatched to legacy module. Shouldn't we maintain this behavior?

**Review by sjchmiela — COMMENTED** [review]
Thank you for digging so deep into this issue and squashing the bug at its source! 🎉

File: android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java (hunk 2 above)
Comment (lukmccall) [reply android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java:71]: We could, but it'll fail with the same error. Why? Cause the old code do the same logic. It tries to get manifest from the database, which doesn't exist.

**Review by sjchmiela — APPROVED** [review]
Superb! 🥇

**Pull request merged**